cilium-cli/install/install.go

```diff
@@ -224,18 +223,14 @@ func (k *K8sInstaller) preinstall(ctx context.Context) error {
 			}
 		}
 	case k8s.KindEKS:
-		chainingMode := getChainingMode(helmValues)
-
-		// Do not stop AWS DS if we are running in chaining mode
-		if chainingMode != "aws-cni" && !k.params.IsDryRun() {
-			if _, err := k.client.GetDaemonSet(ctx, AwsNodeDaemonSetNamespace, AwsNodeDaemonSetName, metav1.GetOptions{}); err == nil {
-				k.Log("🔥 Patching the %q DaemonSet to evict its pods...", AwsNodeDaemonSetName)
-				patch := []byte(fmt.Sprintf(`{"spec":{"template":{"spec":{"nodeSelector":{"%s":"%s"}}}}}`, AwsNodeDaemonSetNodeSelectorKey, AwsNodeDaemonSetNodeSelectorValue))
-				if _, err := k.client.PatchDaemonSet(ctx, AwsNodeDaemonSetNamespace, AwsNodeDaemonSetName, types.StrategicMergePatchType, patch, metav1.PatchOptions{}); err != nil {
-					k.Log("❌ Unable to patch the %q DaemonSet", AwsNodeDaemonSetName)
-					return err
-				}
-			}
+		// detect AWS AMI type
+		if err := k.awsRetrieveNodeImageFamily(); err != nil {
+			return err
+		}
+
+		// setup chaining mode
+		if err := k.awsSetupChainingMode(ctx, helmValues); err != nil {
```

Comment:
This part has only been moved to the aws.go file, correct? No changes impacting the above retrieval of the node image familly.

Reply:
Correct -- I have split it in a separate commit for easier review of the move change.